ggshield/core/scan/scannable.py

```diff
@@ -68,11 +70,27 @@ def is_longer_than(self, max_utf8_encoded_size: int) -> bool:
         """
         raise NotImplementedError
 
-    @property
     @abstractmethod
+    def _read_content(self) -> None:
+        """Read the content of the scannable"""
+        raise NotImplementedError
+
+    @property
     def content(self) -> str:
         """Return the decoded content of the scannable"""
-        raise NotImplementedError
+        if self._content is None:
+            self._read_content()
+        if self._content is not None:
+            return self._content
+        raise ValueError("content is None after reading")
```

Comment:
nit: I would write this like this, it feels more natural to me:

```suggestion
        if self._content is None:
            self._read_content()
            if self._content is None:
                raise ValueError("content is None after reading")
        return self._content
```